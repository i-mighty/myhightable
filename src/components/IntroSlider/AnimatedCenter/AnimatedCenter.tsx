import * as Animatable from 'react-native-animatable';
import styled from 'styled-components/native';

const AnimatedCenter = styled(Animatable.View)`
  margin-top: -70px;
  height: 100%;
  width: 100%;
  max-width: 360px;
  justify-content: center;
  align-items: center;
`;
export default AnimatedCenter;
