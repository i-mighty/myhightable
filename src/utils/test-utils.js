import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
