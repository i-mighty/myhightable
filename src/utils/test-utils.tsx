import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
