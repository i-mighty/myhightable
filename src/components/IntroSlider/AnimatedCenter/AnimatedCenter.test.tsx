import AnimatedCenter from './AnimatedCenter';
import { renderWithProps } from '@src/utils/test-utils';

const defaultProps = {};
const animatedProps = { testID: 'animatedCenter', animation: 'fadeIn' };

describe('AnimatedCenter', () => {
  it('renders properly', () => {
    const { baseElement } = renderWithProps(AnimatedCenter, defaultProps);
    expect(baseElement).toMatchSnapshot();
  });

  it('renders properly with animation', () => {
    const { getByTestId, baseElement } = renderWithProps(
      AnimatedCenter,
      animatedProps,
    );
    const container = getByTestId(animatedProps.testID);
    expect(container.getProp('animation')).toBe(animatedProps.animation);
    expect(baseElement).toMatchSnapshot();
  });
});
