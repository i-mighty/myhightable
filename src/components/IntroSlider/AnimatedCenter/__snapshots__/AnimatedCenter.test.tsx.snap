// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedCenter renders properly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "marginTop": -70,
          "maxWidth": 360,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`AnimatedCenter renders properly with animation 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      animation="fadeIn"
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "marginTop": -70,
          "maxWidth": 360,
          "width": "100%",
        }
      }
      testID="animatedCenter"
    />
  </View>
</View>
`;
